tracing/src/main/java/io/airlift/tracing/OpenTelemetryModule.java

```diff
@@ -0,0 +1,113 @@
+        this.serviceVersion = requireNonNull(serviceVersion, "serviceVersion is null");
+    }
+
+    @Override
+    public void configure(Binder binder)
+    {
+        configBinder(binder).bindConfig(OpenTelemetryConfig.class);
+    }
+
+    @Provides
+    @Singleton
+    public OpenTelemetry createOpenTelemetry(NodeInfo nodeInfo, OpenTelemetryConfig config)
+    {
+        AttributesBuilder attributes = Attributes.builder()
+                .put(ResourceAttributes.SERVICE_NAME, serviceName)
+                .put(ResourceAttributes.SERVICE_VERSION, serviceVersion)
+                .put(ResourceAttributes.SERVICE_INSTANCE_ID, nodeInfo.getNodeId())
+                .put(ResourceAttributes.DEPLOYMENT_ENVIRONMENT, nodeInfo.getEnvironment())
+                .putAll(attribute(ResourceAttributes.PROCESS_RUNTIME_NAME, System.getProperty("java.runtime.name")))
+                .putAll(attribute(ResourceAttributes.PROCESS_RUNTIME_VERSION, System.getProperty("java.runtime.version")))
+                .putAll(attribute(ResourceAttributes.PROCESS_RUNTIME_DESCRIPTION, processRuntime()))
+                .putAll(attribute(ResourceAttributes.OS_TYPE, osType()))
+                .putAll(attribute(ResourceAttributes.OS_NAME, OS_NAME.value()))
+                .putAll(attribute(ResourceAttributes.OS_VERSION, OS_VERSION.value()))
+                .putAll(attribute(ResourceAttributes.HOST_ARCH, hostArch()));
+
+        Resource resource = Resource.getDefault().merge(Resource.create(attributes.build()));
+
+        SpanExporter spanExporter = OtlpGrpcSpanExporter.builder()
+                .setEndpoint(config.getEndpoint())
+                .build();
+
+        SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor.builder(spanExporter).build())
+                .setResource(resource)
+                .build();
+
+        return OpenTelemetrySdk.builder()
+                .setTracerProvider(tracerProvider)
+                .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
+                .build();
+    }
+
+    private static String processRuntime()
+    {
+        String vendor = JAVA_VM_VENDOR.value();
+        String name = JAVA_VM_NAME.value();
+        String version = JAVA_VM_VERSION.value();
+        if ((vendor == null) && (name == null) && (version == null)) {
+            return null;
+        }
+        return "%s %s %s".formatted(vendor, name, version);
+    }
+
+    private static String osType()
+    {
+        return switch (nullToEmpty(OS_NAME.value())) {
+            case "Linux" -> "linux";
+            case "Mac OS X" -> "darwin";
+            default -> null;
+        };
+    }
+
+    private static String hostArch()
+    {
+        return switch (nullToEmpty(OS_ARCH.value())) {
+            case "amd64", "x86_64" -> "amd64";
+            case "aarch64" -> "arm64";
+            case "ppc64le" -> "ppc64";
+            default -> null;
```

Comment:
Is it safe to let this fall through and just pass the denormalized string through if it's a new arch type? I think there are some cloud vendors looking at risc-v machines.

Reply:
According to https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/host/

> host.arch has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.

In order to know if it's not a well known value like `x86`, you'd have to map all the values like `x86_32`, `i686`, etc., so I only passed the known values that we currently support in Trino. Surprisingly, they don't have a value for RISC-V yet.